src/main/java/com/atlassian/bitbucket/jenkins/internal/client/supply/BitbucketCapabilitiesSupplier.java

```diff
@@ -0,0 +1,25 @@
+package com.atlassian.bitbucket.jenkins.internal.client.supply;
+
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketRequestExecutor;
+import com.atlassian.bitbucket.jenkins.internal.model.AtlassianServerCapabilities;
+import com.google.common.base.Supplier;
+import okhttp3.HttpUrl;
+
+public class BitbucketCapabilitiesSupplier implements Supplier<AtlassianServerCapabilities> {
+
+    private final BitbucketRequestExecutor bitbucketRequestExecutor;
+
+    public BitbucketCapabilitiesSupplier(BitbucketRequestExecutor bitbucketRequestExecutor) {
+        this.bitbucketRequestExecutor = bitbucketRequestExecutor;
+    }
+
+    @Override
+    public AtlassianServerCapabilities get() {
+        HttpUrl url =
+                bitbucketRequestExecutor.getBaseUrl().newBuilder()
+                        .addPathSegment("rest")
+                        .addPathSegment("capabilities")
+                        .build();
+        return bitbucketRequestExecutor.makeGetRequest(url, AtlassianServerCapabilities.class).getBody();
```

Comment:
My guava cache knowledge is a bit rusty (which arguably is an advantage given the, uhm, vintage, version used).
Anyway, what about edge cases here? What happens in case of an exception?

Reply:
That's a good question- will add a test for it. There aren't any checked exceptions so I'm going to assume the behaviour we want is the same as it was without the cache- any runtime errors bubble up to the consumer where they can handle them?